ideals.md

```diff
@@ -10,75 +10,52 @@ These ideals also help define the scope of Cockpit. That scope is limited, and n
 
-Cockpit opens one port for browser connections: by default that is 9090. We work well in cases where a bastion host runs Cockpit and connects out to the other servers via SSH.
+When packaging a stable release of Cockpit for release with a given OS, most of the “Cockpit” branding should be removed, and OS specific branding is put in its place.
 
-### Cockpit helps the server evolve coherently
+### Cockpit is zero footprint
 
-When developing Cockpit we run into situations where we notice the underlying server behavior is not coherent. We try to resolve that by contributing in the usual open source fashion, before implementing a hack or work around.
+The job of a server is to serve stuff to others. The tools to configure the server itself should not waste resources needed for that. As such, Cockpit has (as near as makes no difference) zero memory and process footprint on the server when not in use. Disk space usage of Cockpit is minimal and the required set of dependencies is scrutinized.
 
 ### Cockpit is designed first
 
-When developing Cockpit we want to design first and let that drive the implementation. As one would expect, there are often concrete limitations here, but this is the ideal.
-
-### Cockpit is only the project name
-
-Users of Cockpit should feel they are interacting with the underlying Server OS, not with Cockpit.
+When developing Cockpit we want to design first, with real-world use cases driving which features we work on.
 
-Cockpit can look different on different operating systems, because it’s the UI for the OS, and not a external tool.
-
-When packaging a stable release of Cockpit for release with a given OS, most of the “Cockpit” branding should be removed, and OS specific branding is put in its place.
-
-### Cockpit is pluggable
+### Cockpit helps the server evolve coherently
 
-Cockpit allows others to add additional UI pieces. We do not allow arbitrary extension of the UI all over the place, but at distinct extension points.
+When developing Cockpit we run into situations where we notice the underlying server behavior is not coherent. We try to resolve that by contributing in the usual open source fashion, before implementing a hack or work around.
 
 ### Cockpit is tested
 
 Every feature and change in Cockpit is covered by an automatic test. A proposed change only lands in the code base after it passes all tests. Tests run on all [Operating System](running.html) releases on which Cockpit is supported.
 
+In addition, the design of new features and changes in existing features are tested in usability studies.
+
+### Cockpit is backwards-compatible
+
+Cockpit can connect to other instances of itself. These might be much older, because we support many operating systems with different release cadences. It is important that this keeps working, so we try to as little API as possible to `cockpit.js` and no other API.
```

Comment:
Lets also add:

Cockpit's Javascript API in ```cockpit.js``` may not change in an incompatible way. External Cockpit components (or plugins) that use the ```cockpit.js``` javascript API should continue to function despite newer versions of Cockpit being released. 

There is no forseen major incompatible version change for this Javascript API in Cockpit's future.

Reply:
"try to as little API" → this sentence no verb. "have" or "expose"? The rest of the sentence sounds a bit  cumbersome too, so how about "…different release cadences. To achieve that, keep the `cockpit.js` Javascript API minimal, stable, and backwards compatible, and expose no other APIs".

Reply:
Thanks. The current version doesn't include the "keep the API minimal" aspect, but it's sort of implied.